tasklist/client/src/v2/utils/parseDenialReason.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+export const parseDenialReason = async (
+  error: unknown,
+  type: 'assignment' | 'unassignment' | 'completion',
+): Promise<string | undefined> => {
+  const response = (error as {response?: Response})?.response;
+
+  if (!response || response.status !== 409) return undefined;
+
+  const responseJson = await response?.json();
+  const detail = responseJson?.detail as string;
+
+  let deniedReason: string = `The task ${type} was rejected by the system.`;
+  const match = detail.match(/Reason to deny:\s*(.*)/i);
```

Comment:
🔧 I think this will also match the quotes around the actual reason. Do we want to display those? I lean toward removing the quotes from the matched group